### ISSUE TITLE
install from cuda repo fails

### ISSUE THREAD

**Issue by KasperSkytte** [NVIDIA/ansible-role-nvidia-driver#60]
When setting `nvidia_driver_ubuntu_install_from_cuda_repo: yes` I get:

```
TASK [nvidia.nvidia_driver : install driver packages] ****************************************************
fatal: [nodename]: FAILED! => {"changed": false, "msg": "No package matching 'cuda-drivers-510' is available"}
```

The package doesn't exist:
```
$ apt-cache search ^cuda-drivers
cuda-drivers-fabricmanager-450 - Meta-package for FM and Driver
cuda-drivers-fabricmanager-460 - Transitional package for cuda-drivers-fabricmanager-510
cuda-drivers-fabricmanager-470 - Meta-package for FM and Driver
cuda-drivers-fabricmanager-510 - Meta-package for FM and Driver
```

No issues when setting `nvidia_driver_ubuntu_install_from_cuda_repo: no`, but I would like to install from the CUDA repository though.

**Comment by BarthV** [comment]
same probleme here

**Comment by ajdecon** [comment]
I can confirm I'm seeing the same thing. The driver role code hasn't changed since the last successful test, but this looks like an issue in the upstream repo. I'll check with the repo maintainers.

**Comment by ajdecon** [comment]
Confirmed there was an issue with the upstream apt package repository, and this should now be fixed.

Tested via triggering a CI run and confirming that all install paths are successful: https://github.com/NVIDIA/ansible-role-nvidia-driver/runs/6314471044

Please make sure to run `apt-get update` before attempting a new install. Closing this based on the successful test, but feel free to re-open this issue if you see the issue persist.

**Comment by KasperSkytte** [comment]
I have no permission to re-open the issue

**Comment by ajdecon** [comment]
Oops! Re-opening the issue myself and will kick off another test on my end. (both through CI and on a local VM)

**Comment by ajdecon** [comment]
@KasperSkytte : Both the [CI tests](https://github.com/NVIDIA/ansible-role-nvidia-driver/runs/6324432210) and my local VM tests are successfully installing the driver from the CUDA repos. This worked on all of Ubuntu 18.04, Ubuntu 20.04, and CentOS 7.

Can you confirm if you are still seeing this issue?

If so -- can you please do the following to help troubleshoot?

1. On a fresh VM, run the role and provide a gist with your full log
2. On a VM with the updated repos installed, please run `sudo apt-get update` and `sudo apt-get install cuda-drivers-510` manually, and provide a gist with the full log

**Comment by BarthV** [comment]
on my side things are settling down.
driver + docker ansible roles are now doing their jobs on a bare new setup, everything is installed from cuda repo with success.

I think the main problem is dealing on machines where the previous version of this role was running.
In this case i'm afraid :
* ubuntu packages seems to be conflicting
* old nvidia apt keys are conflicting too
* cuda-keyring is not able to be installed

A better care of these cases should at least raise a warning or error in this role

**Comment by BarthV** [comment]
I can help by giving the system state and files created by the "old" setup method.

**Comment by KasperSkytte** [comment]
@ajdecon Thank you for being so persistent. I got it working now too. Turns out my "fresh" VM wasn't so fresh. Was confusing myself with multiple ones. Works out of the box now on Ubuntu 20. I'm closing again.

**Comment by KasperSkytte** [comment]
Agree with @BarthV that a few tasks to clean up from any other method(s) to install nvidia driver would be handy.

**Comment by KurtAhn** [comment]
I was also struggling to install the driver with the Ansible role and faced this exact same issue. So, I followed the [installation guide](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#ubuntu-installation). Because I wasn't able to install the `cuda-keyring` package, I followed the alternative steps described in 3.8.3.*2*, and this allowed me to install the driver. For good measure, I re-ran Ansible after this, and the driver installed successfully. I hope this is useful to someone.
